src/main/java/com/couchbase/intellij/tree/cblite/CBLTreeRightClickListener.java

```diff
@@ -149,62 +216,108 @@ public void actionPerformed(@NotNull AnActionEvent e) {
         AnAction createDocument = new AnAction("Create Document") {
             @Override
             public void actionPerformed(@NotNull AnActionEvent e) {
-                CBLOpenDocumentDialog dialog = new CBLOpenDocumentDialog(true, project, tree, userObject.getScope(), userObject.getText());
+                CBLOpenDocumentDialog dialog = new CBLOpenDocumentDialog(true, project, tree, userObject.getScope(),
+                        userObject.getText());
                 dialog.show();
             }
         };
         actionGroup.add(createDocument);
 
+        AnAction deleteCollection = new AnAction("Delete Collection") {
+            @Override
+            public void actionPerformed(@NotNull AnActionEvent e) {
+                int result = Messages.showYesNoDialog("<html>Are you sure you want to delete the collection <strong>"
+                        + userObject.getText() + "</strong>?</html>", "Delete Collection", Messages.getQuestionIcon());
+                if (result != Messages.YES) {
+                    return;
+                }
+
+                try {
+
+                    if (ActiveCBLDatabase.getInstance() != null
```

Comment:
Can you give an example of when ActiveCBLDatabase.getInstance() is null? Why did you add this check?

Reply:
removed it